profiling/src/php_ffi.c

```diff
@@ -146,3 +146,68 @@ zend_execute_data* ddog_php_prof_get_current_execute_data()
 {
     return EG(current_execute_data);
 }
+
+#if CFG_STACK_WALKING_TESTS
+zend_execute_data* ddog_php_test_create_fake_zend_execute_data(int depth) {
+    if (depth <= 0 || depth > 99) {
+        return NULL;
+    }
+
+    zend_execute_data *execute_data = (zend_execute_data *) malloc(sizeof(zend_execute_data));
+    memset(execute_data, 0, sizeof(zend_execute_data));
+
+    execute_data->prev_execute_data = ddog_php_test_create_fake_zend_execute_data(depth - 1);
+
+    // add function name to stack frame
+    zend_string *func_name = malloc(sizeof(zend_string) + 16);
+    func_name->h = 0;
+    func_name->len = 16;
+    memcpy(func_name->val, "function name 00", 16);
+    if (depth > 9) {
```

Comment:
snprintf seems like less work ^^

Reply:
I actually found a way using `dlsym()` to look for the original symbol, CI currently running